server/controllers/profile.js

```diff
@@ -0,0 +1,44 @@
+const Profile = require("../models/Profile");
+const User = require("../models/User");
+const asyncHandler = require("express-async-handler");
+
+// @route POST /profile/edit
+// @desc edit user profile
+// @access Public
+exports.editProfile = asyncHandler(async (req, res, next) => {
+  const user = await User.findById(req.user.id);
+
+  if (!user) {
+    res.status(404);
+    throw new Error("User doesn't exist");
+  }
+  try {
+    user.profile.set(req.body);
+    const updated_user = await user.save();
```

Comment:
```suggestion
    const updatedUser = await user.save();
```
Please use camelCase as it is the standard naming convention for JS